geomstats/stiefel.py

```diff
@@ -0,0 +1,198 @@
+"""
+Stiefel manifold St(p,n),
+a set of all orthonormal p-frames in n-dimensional space,
+where p <= n
+"""
+
+import logging
+import random
+
+import geomstats.backend as gs
+
+from geomstats.embedded_manifold import EmbeddedManifold
+from geomstats.riemannian_metric import RiemannianMetric
+
+TOLERANCE = 1e-6
+EPSILON = 1e-6
+
+class Stiefel(EmbeddedManifold):
+
+    def __init__(self, p, n):
+        self.p = p
+        self.n = n
+
+        self.dimension = int(p * n - (p * (p - 1) / 2))
+
+    def belongs(self, point, tolerance=TOLERANCE):
+        """
+        Evaluate if a point belongs to St(p,n),
+        i.e. if it is a p-frame in n-dimensional space,
+        and it is orthonormal.
+        """
+        point = gs.to_ndarray(point, to_ndim=3)
+        (_, point_dim) = point.shape
```

Comment:
You don't need the parentheses here.

Reply:
Btw: doesn't this throw an error? "Too many values to unpack", since point.shape is (n_points, n, p)?

Can you add the unit tests?

Reply:
I let you address this ^